### ISSUE TITLE
How i can check for missing types ?

### ISSUE THREAD

**Issue by Persovt** [mscharley/generic-type-guard#509]

**Comment by mscharley** [comment]
You'll need to provide a bit more context than this, like an example you're trying to prevent.

**Comment by Persovt** [comment]
Если у меня type guard говорит, что тип отличается, как я могу понять, что в типе отличается ?

**Comment by mscharley** [comment]
The TL;DR is that this library doesn't support showing where the error is due to limitations in TypeScripts type guard syntax. The basic format for a TypeScript type guard is `(value: unknown): value is T => [some check here]` and that return type can't be wrapped in any extra context, the boolean return value is all that the function can return.

The longer version is that this is a bit of a nightmare feature to support well, no matter what limitations you're under. JSON schema via `ajv` and `runtypes` are two alternatives I'd consider if this is a hard requirement for you. Each has their own benefits. `generic-type-guard`s main appeal is that it is simple, works directly with the language and since it just relies on function composition, it's fully compatible with anything else you might choose to use. GTG guards and Runtypes are actually fully compatible with each other, in both directions.

First, some options with GTG:

1. Log the value you are testing and compare manually. I've honestly had pretty good luck with this option, unless your inputs are massive. This is often my actual go-to even when using other libraries with better error reporting.
2. If your inputs are large, then one option is to use `isUnknown` to flag sections of your type to be processed in a second stage. This means that each failure is considering a smaller portion of the data which should be easier to compare manually, and depending on exactly how big the data is this could even be considered a potential performance optimisation as well. This works particularly well for arrays, you can use `narrowArray` to check an `unknown[]` into `T[]` given an appropriate typeguard for `T`.

Some limitations I've run into with the algorithms for detecting what exactly is different include:

1. Unions. Both JSON schema and `runtypes` both now support discriminated unions to help with this, but in general it's impossible to tell which branch of a union to go down for the validation error without some kind of discriminator, so the first union you run into will be the level you always get errors from. Personally, I deal with a lot of discriminated unions and before this was a feature I ran into a lot of issues with it just giving an error saying there was an error somewhere in the whole type anyway.
2. Custom logic. Similar issue to unions, if you need any kind of custom checking logic that the library doesn't support first-party then you're out of luck. JSON schema doesn't support it at all on account of being defined in JSON, and `runtypes` supports custom logic but if there is an error in it then the custom error reporting will return at the level of your custom logic and say "there's an error somewhere in there".

**Comment by Persovt** [comment]
Okay, thank you. I'll select "Runtypes" maybe. In my case I would like set type for "Axios" request  an unknown api. I would like get request and validate result them.
